master/addtrigger.sql

```diff
@@ -72,32 +76,17 @@ AFTER INSERT OR UPDATE OF
     state, start_time, end_time, runner_state, restarts, tags
 ON trials
 FOR EACH ROW EXECUTE PROCEDURE stream_trial_change();
--- DELETE should fire BEFORE to guarantee the experiment still exists to grab the project_id.
+-- DELETE should fire BEFORE to guarantee the experiment still exists to grab the workspace_id.
 DROP TRIGGER IF EXISTS stream_trial_trigger_d ON trials;
 CREATE TRIGGER stream_trial_trigger_d
 BEFORE DELETE ON trials
 FOR EACH ROW EXECUTE PROCEDURE stream_trial_change();
 
--- Trigger for detecting trial ownership changes derived from experiments.project_id.
-CREATE OR REPLACE FUNCTION stream_trial_ownership() RETURNS TRIGGER AS $$
+-- trigger function to notify when permission changes are detected.
```

Comment:
when we promote `addtrigger.sql` into a proper migration (which is probably a good idea soonish, doesn't have to be this pr), we should take one of the following approaches:

- leave a comment here explaining that we define the trigger in OSS but only activate it on EE tables, so that's why it isn't touched here
- move the trigger function definition to the SQL migration in EE that calls the trigger

I'm honestly not sure which is best.  But the current state is a bit confusing to the reader.

Reply:
I feel like I have a slight bias towards defining the trigger in OSS - I dislike splitting migrations for EE and I don't think that it would compromise EE.

Reply:
I have mixed feelings.

**Moving to EE**
_Pros:_
- Less confusion from a migration perspective 
- Prevents unintentional use of this channel
- Gives us the option to remove the `permissionChangeListener` code path from OSS
   - Since OSS doesn't restrict access to viewing planned streamable entities
   - We'd still have to handle moving experiments & projects
   
_Cons:_
- Requires split handling of pq.Listener.Listen() between OSS & EE (at least for this channel), since a failure due to the channel not existing intentionally vs an actual failure will look identical (if we leave `permissionChangeListener`)
 ____
**Leaving in OSS**
_Pros:_
- Consistent code between OSS & EE for monitoring permission changes, whether they are actually used or not
- Less of a lift from current state (still requires handling when experiments & projects move)

_Cons:_
- Misleading pq.Listener & trigger function in OSS
- Confusing `addtrigger.sql`  (would need to be made clearer prior to/when we make this a formal migration)

Reply:
My preference is to move it into EE + moving the logic for monitoring permission changes into the authz package that gets called where it currently exists (where the basic implementation returns some form of a nil/noop)

Reply:
Moved the logic for detecting permission changes into the authz package and that gets called in the same spot in websocket. This allows use to abstract any of the implementation details out of system.go, and handle permission changes based on the authz provider.

This also means that we only need the permission change channel in EE